src/services/item/repository.ts

```diff
@@ -198,6 +200,38 @@ export class ItemRepository {
       query.andWhere('item.type IN (:...types)', { types });
     }
 
+    const allKeywords = params?.keywords?.filter((s) => s && s.length);
+    if (allKeywords?.length) {
+      const keywordsString = allKeywords.join(' ');
+      query.andWhere((q) => {
+        // search in english by default
+        q.where("item.search_document @@ plainto_tsquery('english', :keywords)", {
+          keywords: keywordsString,
+        });
+
+        // no dictionary
+        q.orWhere("item.search_document @@ plainto_tsquery('simple', :keywords)", {
+          keywords: keywordsString,
+        });
+
+        // raw words search
+        allKeywords.forEach((k, idx) => {
+          q.orWhere(`item.name ILIKE :k_${idx}`, {
+            [`k_${idx}`]: `%${k}%`,
+          });
+        });
+
+        // search by member lang
+        const memberLang = actor?.lang;
+        if (memberLang && memberLang != 'en' && ALLOWED_SEARCH_LANGS[memberLang]) {
+          q.orWhere('item.search_document @@ plainto_tsquery(:lang, :keywords)', {
+            keywords: keywordsString,
+            lang: ALLOWED_SEARCH_LANGS[memberLang],
+          });
+        }
```

Comment:
I'm wondering if you really need the member for that. What about directly passing the language in parameter ?

Reply:
We don't need the full member, but I prefer passing the full entity so it's similar to `accessible` + if one day we need an additional data, it's already there.